### PR TITLE
Replace django-password-reset with Django's password reset views

### PR DESCRIPTION
**Pull request by PhilippMatthes**
Fixes: #204 

*Following changes were made:*

- Remove django-password-reset from `Pipfile` and update `Pipfile.lock`
- Remove all views using django-password-reset
- Remove the View `PasswordChangeView` since Django supports this out of the box since 1.11
- Replace `PasswordChangeView` with `auth.views.PasswordChangeView`

- Add `auth.views.PasswordChangeDoneView` to display a message on password change success

- Replace `password_reset.Recover` with `auth.views.PasswordResetView`
- Replace `password_reset.RecoverDone` with `auth.views.PasswordResetDoneView`
- Replace `password_reset.Reset` with `auth.views.PasswordResetConfirmView`
- Replace `password_reset.ResetDone` with `auth.views.PasswordResetCompleteView`
Additionally,
- Adjust password reset url schemes
- Adjust existing password reset templates to fit new parameters passed by the Django views

- Add new integration tests for the password reset workflow

TODO: @martinmo could you please pick all needed changes in `Pipfile.lock`? I feel like I'd damage something. 

Other than that, I will already request review, because other than the needed changes in `Pipfile.lock`, there currently is nothing left to do.

### CODE REVIEW

**Comment by PhilippMatthes** [comment]
The builds are not succeeding, because there are some odd problems with the Pipenv that should fix themselves when the correct changes are made to `Pipfile` and `Pipfile.lock`

**Comment by martinmo** [comment]
Really awkward errors. The hashes of the packages changed on PyPI, which should not happen. This also happens on the `master` branch with unchanged `Pipfile` etc., if I start with a fresh virtualenv and do a `pipenv install --dev`. Need to investigate this further tomorrow, until then everything is put on hold. The last cron build on Travis was 5 days ago and succeeded. I restarted that build and it still builds fine... maybe the "good" stuff is still cached?!

**Comment by PhilippMatthes** [comment]
> Tests are missing ⚠️

I added new integration tests to test our password reset workflow.

**Comment by martinmo** [comment]
I sorted the Pipfile problems in another branch that is currently building on Travis. Once that's merged into master you can rebase your PR. Revert the commit that changed the `Pipfile*` and run `pipenv uninstall django-password-reset`.

**Comment by martinmo** [comment]
> Once that's merged into master you can rebase your PR. Revert the commit that changed the Pipfile* and run pipenv uninstall django-password-reset.

As discussed on Telegram, I've done this already for you and the Travis build should now succeed. I will review your changes tomorrow.

File: tests/accounts/tests.py [hunk 1]
```diff
@@ -196,3 +196,62 @@ def test_signup_workflow(self):
 
         self.assertTrue(self.client.login(username="bob", password="secret"),
                         "Login should succeed after activation")
+
+
+class PasswordRecorverWorkflowTest(SimpleAccountsData, TestCase):
```
Comment (martinmo) [inline tests/accounts/tests.py]: There's a typo in the class name

```
class PasswordRecorverWorkflowTest
                  ^----- here
```

File: tests/accounts/tests.py [hunk 2]
```diff
@@ -196,3 +196,62 @@ def test_signup_workflow(self):
...
+        current_location, status_code = response.redirect_chain[0]
+        self.assertEquals(status_code, 302)
+
+        form_data = {
+            "new_password1": "ji32k7au4a83",
+            "new_password2": "ji32k7au4a83"
+        }
+        response = self.client.post(current_location, data=form_data, follow=True)
+        self.assertContains(response, "Your new password has been saved.")
+
```
Comment (martinmo) [inline tests/accounts/tests.py]: Please add an assert that verifies that a second "click" on the activation link fails with the message that it's not valid anymore (becuase already used).

File: inloop/accounts/views.py [hunk 3]
```diff
@@ -107,15 +86,36 @@ def get_success_url(self, user):
...
+    template_name="password_reset/reset_sent.html"
+)
+
+password_reset_confirm = PasswordResetConfirmView.as_view(
+    success_url=reverse_lazy("accounts:password_reset_complete"),
+    template_name="password_reset/reset.html"
+)
+password_reset_complete = PasswordResetCompleteView.as_view(
+    template_name="password_reset/recovery_done.html"
+)
```
Comment (martinmo) [inline inloop/accounts/views.py]: I would prefer to stick to the default `template_name` values that are given in https://docs.djangoproject.com/en/1.11/topics/auth/default/#all-authentication-views.

File: inloop/accounts/views.py [hunk 4]
```diff
@@ -1,38 +1,21 @@
...
 from constance import config
-from password_reset.views import Recover, RecoverDone, Reset, ResetDone
 from registration.backends.hmac.views import (ActivationView as HmacActivationView,
                                               RegistrationView as HmacRegistrationView)
 
 from inloop.accounts.forms import SignupForm, StudentDetailsForm, UserChangeForm
 from inloop.accounts.models import StudentDetails
 
 
-class PasswordChangeView(LoginRequiredMixin, FormView):
```
Comment (martinmo) [inline inloop/accounts/views.py]: Why did you remove this class? Can you verify that Django's `PasswordChangeView` mixes in `LoginRequiredMixin`?

Comment (PhilippMatthes) [reply inloop/accounts/views.py]: https://github.com/django/django/blob/0b8abd7cdf1a5bae96dd0640af10ad504f104d06/django/contrib/auth/views.py#L336

Django's `PasswordChangeView` redirects users who are not logged in back to the login page.

File: inloop/accounts/views.py (hunk 3 above)
Comment (PhilippMatthes) [reply inloop/accounts/views.py]: Ok, I'll change the template names.

File: tests/accounts/tests.py (hunk 1 above)
Comment (PhilippMatthes) [reply tests/accounts/tests.py]: Oops, yep, will change that.

File: tests/accounts/tests.py (hunk 2 above)
Comment (PhilippMatthes) [reply tests/accounts/tests.py]: You are right, this should be added to the test.
